internal/operator-controller/rukpak/bundle/config.go

```diff
@@ -6,119 +6,498 @@ import (
+}
+
+// validateK8sNamespace checks if a string is a valid Kubernetes namespace name.
+func validateK8sNamespace(value interface{}) error {
+	str, ok := value.(string)
+	if !ok {
+		return fmt.Errorf("value must be a string")
+	}
+
+	errs := validation.IsDNS1123Subdomain(str)
+	if len(errs) > 0 {
+		return fmt.Errorf("%s", errs[0])
 	}
 
 	return nil
 }
 
-// isWatchNamespaceConfigSupported returns true when the bundle exposes a watchNamespace configuration. This happens when:
-// - SingleNamespace and/or OwnNamespace install modes are supported
-func isWatchNamespaceConfigSupported(bundleInstallModeSet sets.Set[v1alpha1.InstallMode]) bool {
-	return bundleInstallModeSet.HasAny(
+// isWatchNamespaceConfigurable answers: Can the user specify which namespace to watch?
+//
+//   - When bundle supports SingleNamespace -> YES (user specifies a namespace different from install namespace)
+//   - When bundle supports OwnNamespace -> YES (user must explicitly set to install namespace)
+//   - When bundle supports only AllNamespaces -> NO (watches all namespaces, no config needed)
+//   - When bundle supports AllNamespaces + OwnNamespace -> YES (user can optionally specify)
+func isWatchNamespaceConfigurable(installModes sets.Set[v1alpha1.InstallMode]) bool {
+	return installModes.HasAny(
 		v1alpha1.InstallMode{Type: v1alpha1.InstallModeTypeSingleNamespace, Supported: true},
 		v1alpha1.InstallMode{Type: v1alpha1.InstallModeTypeOwnNamespace, Supported: true},
 	)
 }
 
-// isWatchNamespaceConfigRequired returns true if the watchNamespace configuration is required. This happens when
-// AllNamespaces install mode is not supported and SingleNamespace and/or OwnNamespace is supported
-func isWatchNamespaceConfigRequired(bundleInstallModeSet sets.Set[v1alpha1.InstallMode]) bool {
-	return isWatchNamespaceConfigSupported(bundleInstallModeSet) &&
-		!bundleInstallModeSet.Has(v1alpha1.InstallMode{Type: v1alpha1.InstallModeTypeAllNamespaces, Supported: true})
+// isWatchNamespaceConfigRequired answers: Must the user set watchNamespace?
+//
+//   - When bundle supports only OwnNamespace -> REQUIRED (must set it)
+//   - When bundle supports only SingleNamespace -> REQUIRED (must set it)
+//   - When bundle supports OwnNamespace + SingleNamespace -> REQUIRED (must set it)
+//   - When bundle supports AllNamespaces + OwnNamespace -> OPTIONAL (can default to AllNamespaces)
+//   - When bundle supports only AllNamespaces -> NOT APPLICABLE (no watchNamespace field)
+func isWatchNamespaceConfigRequired(installModes sets.Set[v1alpha1.InstallMode]) bool {
+	return isWatchNamespaceConfigurable(installModes) &&
+		!installModes.Has(v1alpha1.InstallMode{Type: v1alpha1.InstallModeTypeAllNamespaces, Supported: true})
 }
 
-// formatUnmarshalError format JSON unmarshal errors to be more readable
+// formatWatchNamespaceError formats watchNamespace-specific validation errors.
+func formatWatchNamespaceError(msg string, hasWatchNamespace bool, watchNamespaceValue string) error {
```

Comment:
I don't think we should have special handling for a specific field. Bundle configuration is meant to be opaque. E.g. another bundle can come along in the future with a watchNamespace configuration that has a completely different semantic.